compiler/rustc_middle/src/mir/syntax.rs

```diff
@@ -1209,12 +1209,14 @@ pub enum AggregateKind<'tcx> {
     Generator(DefId, SubstsRef<'tcx>, hir::Movability),
 }
 
-#[derive(Copy, Clone, Debug, PartialEq, Eq, TyEncodable, TyDecodable, Hash, HashStable)]
+#[derive(Clone, Debug, PartialEq, Eq, TyEncodable, TyDecodable, Hash, HashStable)]
 pub enum NullOp {
     /// Returns the size of a value of that type
     SizeOf,
     /// Returns the minimum alignment of a type
     AlignOf,
+    /// Returns the offset of a field
+    OffsetOf(Vec<Field>),
```

Comment:
Is there any way to avoid the bump to 48bytes for `Rvalue`? By making this a `Box<[Field]>` or a `ty::List<Field>` (better)?

Reply:
As discussed in https://github.com/rust-lang/rust/pull/106934#discussion_r1125704921, this should be interned, yes.